src/sidebar/components/stream-content.js

```diff
@@ -69,6 +60,11 @@ function StreamContentController(
   store.setSortKey('Newest');
 
   this.loadMore = fetch;
+
+  this.$onInit = () => {
+    this.search.query = () => $routeParams.q || '';
+    this.search.update = q => $location.search({ q });
+  };
```

Comment:
This is the same logic as before, but I just moved accesses of `this.search` into `$onInit`.